### PR TITLE
Implemented alert_notification management

### PR DESCRIPTION
**Pull request by mvisonneau**
This MR allows the configuration of the alert notifications.
Before merging I need to rebase after doing a `govendor fetch github.com/apparentlymart/go-grafana-api` once this MR onto the Grafana go library will have been merged : https://github.com/apparentlymart/go-grafana-api/pull/1

In the meantime feel free to let me know if I forgot something else, mandatory, in order to get this merged.

ie: tests works locally using the updated library

### CODE REVIEW

**Comment by mvisonneau** [comment]
Manual acceptance testing :

```
~$ cat grafana.tf
resource "grafana_alert_notification" "test" {
  type          = "email"
  name          = "test"

  settings {
    "addresses" = "foo@bar.test"
  }
}

~$ terraform -v
Terraform v0.10.0

~$ terraform init

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

~$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed. Cyan entries are data sources to be read.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

  + grafana_alert_notification.test
      is_default:         "false"
      name:               "test"
      settings.%:         "1"
      settings.addresses: "foo@bar.test"
      type:               "email"


Plan: 1 to add, 0 to change, 0 to destroy.


~$ terraform apply
grafana_alert_notification.test: Creating...
  is_default:         "" => "false"
  name:               "" => "test_notification"
  settings.%:         "0" => "1"
  settings.addresses: "" => "foo@bar.test"
  type:               "" => "email"
grafana_alert_notification.test: Creation complete (ID: 5)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

~$ curl -s -X GET https://grafana.example.net/api/alert-notifications/5 | jq
{
  "id": 5,
  "name": "test_notification",
  "type": "email",
  "isDefault": false,
  "settings": {
    "addresses": "foo@bar.test"
  },
  "created": "2017-08-15T17:08:22Z",
  "updated": "2017-08-15T17:08:22Z"
}


~$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

grafana_alert_notification.test: Refreshing state... (ID: 5)
No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, Terraform
doesn't need to do anything.

~$ terraform destroy
grafana_alert_notification.test: Refreshing state... (ID: 5)

The Terraform destroy plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning.
Resources shown in red will be destroyed.

  - grafana_alert_notification.test


Do you really want to destroy?
  Terraform will delete all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

grafana_alert_notification.test: Destroying... (ID: 5)
grafana_alert_notification.test: Destruction complete

Destroy complete! Resources: 1 destroyed.

~$ curl -s -X GET https://grafana.example.net/api/alert-notifications/5 | jq
null
```

**Comment by mvisonneau** [comment]
API MR has been merged, thanks @apparentlymart ! I rebased, tests are now ok 👌

File: grafana/resource_alert_notification.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+			"settings": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func CreateAlertNotification(a *schema.ResourceData, meta interface{}) error {
```
Comment (grubernaut) [inline grafana/resource_alert_notification.go]: Minor Nit: Conventionally we use `d` for `*schema.ResourceData` inside of the CRUD functions.

File: grafana/resource_alert_notification.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
+package grafana
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
```
Comment (grubernaut) [inline grafana/resource_alert_notification.go]: Extra space here

File: grafana/resource_alert_notification.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	if err != nil {
+		return err
+	}
+
+	a.SetId(strconv.FormatInt(id, 10))
+
+	return ReadAlertNotification(a, meta)
+}
+
+func UpdateAlertNotification(a *schema.ResourceData, meta interface{}) error {
```
Comment (grubernaut) [inline grafana/resource_alert_notification.go]: Same as above

File: grafana/resource_alert_notification.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+	alertNotification, err := makeAlertNotification(a)
+	if err != nil {
+		return err
+	}
+
+	return client.UpdateAlertNotification(alertNotification)
+}
+
+func ReadAlertNotification(a *schema.ResourceData, meta interface{}) error {
```
Comment (grubernaut) [inline grafana/resource_alert_notification.go]: Same as above

**Review by grubernaut — CHANGES_REQUESTED** [review]
Mostly looks good! Thanks for the contribution, just a few minor nits and changes.

File: grafana/resource_alert_notification.go (hunk 2 above)
Comment (mvisonneau) [reply grafana/resource_alert_notification.go]: fixed!

File: grafana/resource_alert_notification.go (hunk 1 above)
Comment (mvisonneau) [reply grafana/resource_alert_notification.go]: fixed!

File: grafana/resource_alert_notification.go (hunk 3 above)
Comment (mvisonneau) [reply grafana/resource_alert_notification.go]: fixed!

File: grafana/resource_alert_notification.go (hunk 4 above)
Comment (mvisonneau) [reply grafana/resource_alert_notification.go]: fixed!

**Comment by mvisonneau** [comment]
thanks @grubernaut, fixed the few comments!

**Review by grubernaut — APPROVED** [review]

**Pull request merged**